src/search_options.hpp

```diff
@@ -110,160 +109,112 @@ struct LambdaOptions : public SharedOptions
+    double          preScoringThresh    = 2.0;
 
     LambdaOptions() :
         SharedOptions()
     {
     }
 };
 
-ArgumentParser::ParseResult
-parseCommandLine(LambdaOptions & options, int argc, char const ** argv)
+void parseCommandLine(LambdaOptions & options, int argc, char const ** argv)
 {
     // save commandLine
     for (int i = 0; i < argc; ++i)
         options.commandLine += std::string(argv[i]) + " ";
     eraseBack(options.commandLine);
 
-    std::string programName = "lambda2 " + std::string(argv[0]);
+    std::string programName = "lambda3 " + std::string(argv[0]);
 
     // this is important for option handling:
     if (std::string(argv[0]) == "searchn")
         options.blastProgram = BlastProgram::BLASTN;
 
-    ArgumentParser parser(programName);
+    seqan3::argument_parser parser(programName, argc, argv);
+
     // Set short description, version, and date.
-    setShortDescription(parser, "the Local Aligner for Massive Biological DatA");
+    parser.info.short_description = "the Local Aligner for Massive Biological DatA";
 
     // Define usage line and long description.
-    addUsageLine(parser, "[\\fIOPTIONS\\fP] \\fI-q QUERY.fasta\\fP "
-                         "\\fI-i INDEX.lambda\\fP "
-                         "[\\fI-o output.m8\\fP]");
+    parser.info.synopsis.push_back("[\\fIOPTIONS\\fP] \\fI-q QUERY.fasta\\fP \\fI-i INDEX.lambda\\fP [\\fI-o output.m8\\fP]");
 
     sharedSetup(parser);
 
-#ifndef SEQAN_DISABLE_VERSION_CHECK
-    // version checker initiated by top-level arg parser
-    setDefaultValue(parser, "version-check", "0");
-    hideOption(parser, "version-check");
-#endif
+    // TODO version check
 
-    addOption(parser, ArgParseOption("v", "verbosity",
-        "Display more/less diagnostic output during operation: 0 [only errors]; 1 [default]; 2 "
-        "[+run-time, options and statistics].",
-        ArgParseArgument::INTEGER));
-    setDefaultValue(parser, "verbosity", "1");
-    setMinValue(parser, "verbosity", "0");
-    setMaxValue(parser, "verbosity", "2");
-
-    addSection(parser, "Input Options");
-    addOption(parser, ArgParseOption("q", "query",
-        "Query sequences.",
-        ArgParseArgument::INPUT_FILE,
-        "IN"));
-    setValidValues(parser, "query", getFileExtensions(SeqFileIn()));
-    setRequired(parser, "q");
+    parser.add_option(options.verbosity, 'v', "verbosity", "Display more/less diagnostic output during operation: "
+        "0 [only errors]; 1 [default]; 2 [+run-time, options and statistics].",
+        seqan3::option_spec::DEFAULT, seqan3::arithmetic_range_validator{0, 2});
 
-    if (options.blastProgram != BlastProgram::BLASTN)
+    parser.add_section("Input options");
+
+    // TODO Better solution for file extensions
+    parser.add_option(options.queryFile, 'q', "query", "Query sequences.", seqan3::option_spec::REQUIRED,
+        seqan3::path_existence_validator() | seqan3::file_ext_validator({"fa", "fasta", "fq", "fastq"}));
```

Comment:
Did you update the seqan3 in the repo so that the path-changes are tracked?

Reply:
done now